Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSDetEffCorrVana.py

```diff
@@ -0,0 +1,244 @@
+# pylint: disable=no-init,invalid-name
+import mantid.simpleapi as api
+from mantid.api import PythonAlgorithm, AlgorithmFactory, MatrixWorkspaceProperty
+from mantid.kernel import Direction
+import numpy as np
+
+
+class DNSDetEffCorrVana(PythonAlgorithm):
+    """
+    Peforms detector efficiency correction on a given data workspace
+    using the Vanadium data.
+    This algorithm is written for the DNS @ MLZ,
+    but can be used for other instruments if needed.
+    """
+    properties_to_compare = ['deterota', 'wavelength', 'slit_i_left_blade_position',
+                             'slit_i_right_blade_position', 'slit_i_lower_blade_position',
+                             'slit_i_upper_blade_position', 'polarisation', 'flipper']
+    dataws = None
+    outws_name = None
+    vanaws = None
+    bkgws = None
+    vana_mean_name = None
+
+    def category(self):
+        """
+        Returns category
+        """
+        return 'PythonAlgorithms\\MLZ\\DNS\\CorrectionFunctions\\EfficiencyCorrections'
+
+    def name(self):
+        """
+        Returns name
+        """
+        return "DNSDetEffCorrVana"
+
+    def summary(self):
+        return " Peforms detector efficiency correction \
+                on a given data workspace using the Vanadium data."
+
+    def PyInit(self):
+        self.declareProperty(MatrixWorkspaceProperty("InputWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with experimental data from sample.")
+        self.declareProperty(MatrixWorkspaceProperty("OutputWorkspace", "", direction=Direction.Output),
+                             doc="A workspace to save the corrected data.")
+        self.declareProperty(MatrixWorkspaceProperty("VanaWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with Vanadium data.")
+        self.declareProperty(MatrixWorkspaceProperty("BkgWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with background for Vanadium data.")
+        self.declareProperty("VanadiumMean", "", direction=Direction.Input,
+                             doc="A 2x1 matrix workspace with mean Vanadium counts (optional).")
+
+        return
+
+    def _dimensions_valid(self):
+        """
+        Checks validity of the workspace dimensions:
+        all given workspaces must have the same number of dimensions
+        and the same number of histograms
+        and the same number of bins
+        """
+        nd = self.dataws.getNumDims()
```

Comment:
I think it's not an issue here, but in principle variable names shorter than 3 characters are discouraged (http://www.mantidproject.org/How_to_fix_Pylint_Warnings#Naming) and pylint is set up to catch you.
I guess that was the reason to use the pylint disable for `invalid-name`?

By the way if you want to do the pylint checks locally on your build machine you can build the target `pylintcheck`. Unfornutately I think that's not working well with MSVC at the moment, but on osx and linux it should work well.